### PR TITLE
backtrack code

### PR DESCRIPTION
**Pull request by SamYang10086**

### CODE REVIEW

**Review by SamYang10086 — COMMENTED** [review]
wait for reviews

File: Travel-Planner/Travel-Planner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,6 +7,10 @@
 	objects = {
 
 /* Begin PBXBuildFile section */
+		83EF88DC2CFD962C00D63DC0 /* DateCellView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 83EF88D92CFD962C00D63DC0 /* DateCellView.swift */; };
+		83EF88DD2CFD962C00D63DC0 /* CustomCalendarView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 83EF88D82CFD962C00D63DC0 /* CustomCalendarView.swift */; };
+		83EF88DE2CFD962C00D63DC0 /* DateRangeSelection.swift in Sources */ = {isa = PBXBuildFile; fileRef = 83EF88DA2CFD962C00D63DC0 /* DateRangeSelection.swift */; };
+		83EF88E12CFD983F00D63DC0 /* TripCardView.swift in Sources */ = {isa = PBXBuildFile; fileRef = 83EF88E02CFD983F00D63DC0 /* TripCardView.swift */; };
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner.xcodeproj/project.pbxproj:13]: Do not merge this file

File: Travel-Planner/.DS_Store
Comment (Topia99) [inline Travel-Planner/.DS_Store:1]: not this file

File: .DS_Store
Comment (Topia99) [inline .DS_Store:1]: not this file

File: Travel-Planner/Travel-Planner.xcodeproj/project.xcworkspace/xcuserdata/siyeyang.xcuserdatad/UserInterfaceState.xcuserstate
Comment (Topia99) [inline Travel-Planner/Travel-Planner.xcodeproj/project.xcworkspace/xcuserdata/siyeyang.xcuserdatad/UserInterfaceState.xcuserstate:1]: not this file

File: Travel-Planner/Travel-Planner.xcodeproj/xcuserdata/siyeyang.xcuserdatad/xcschemes/xcschememanagement.plist
Comment (Topia99) [inline Travel-Planner/Travel-Planner.xcodeproj/xcuserdata/siyeyang.xcuserdatad/xcschemes/xcschememanagement.plist:1]: what does this do?

File: Travel-Planner/Travel-Planner/.DS_Store
Comment (Topia99) [inline Travel-Planner/Travel-Planner/.DS_Store:1]: not this file

File: Travel-Planner/Travel-Planner/Assets.xcassets/.DS_Store
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Assets.xcassets/.DS_Store:1]: not this file

File: Travel-Planner/Travel-Planner/CoreData/TripEntity+CoreDataProperties.swift [hunk 2]
```diff
@@ -16,9 +16,9 @@ extension TripEntity {
         return NSFetchRequest<TripEntity>(entityName: "TripEntity")
     }
 
-    @NSManaged public var endDate: Date
-    @NSManaged public var startDate: Date
-    @NSManaged public var title: String
+    @NSManaged public var endDate: Date?
+    @NSManaged public var startDate: Date?
+    @NSManaged public var title: String?
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/CoreData/TripEntity+CoreDataProperties.swift]: why optional?

File: Travel-Planner/Travel-Planner/Screens/TripListView.swift [hunk 3]
```diff
@@ -10,22 +10,73 @@ import SwiftUI
...
         NavigationStack {
-            ScrollView{
+            ScrollView {
                 LazyVStack {
                     ForEach(vm.trips) { trip in
-                        NavigationLink(destination: TripView(vm: DayPlanViewModel(trip: trip))) {
-                            Text(trip.title)
+                        if isEditing {
+                            // Deletable trip row
+                            HStack {
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/TripListView.swift]: is there a way to make this into a list? SwiftUI has system function to edit and delete item in a list

File: Travel-Planner/Travel-Planner/Screens/TripView.swift [hunk 4]
```diff
@@ -35,7 +35,7 @@ struct TripView: View {
             .animation(.easeInOut(duration: 0.3), value: selectedDayPlanIndex) // Slide animation when switch to different day.
             .frame(maxWidth: .infinity, maxHeight: .infinity)
         }
-        .navigationTitle(vm.trip.title)
+        .navigationTitle(vm.trip.title!)
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/Screens/TripView.swift]: make trip.title non optional. Don't force open it

File: Travel-Planner/Travel-Planner/ViewModels/TripViewModel.swift [hunk 5]
```diff
@@ -32,13 +32,14 @@ class TripViewModel: ObservableObject {
         newTrip.title = title
         newTrip.startDate = startDate
         newTrip.endDate = endDate
-        
+
         // Add Empty DayPlan to new trip
         addEmptyDayPlanToNewTrip(newTrip: newTrip, startDate: startDate, endDate: endDate)
```
Comment (Topia99) [inline Travel-Planner/Travel-Planner/ViewModels/TripViewModel.swift:37]: if no changes, don't push

**Review by Topia99 — COMMENTED** [review]
reviwed by Jason

**Review by SamYang10086 — COMMENTED** [review]
....

**Pull request closed without merging**